### ISSUE TITLE
Update Generic Guns mod to be synchronized with current core content

### ISSUE THREAD

**Issue by esotericist** [CleverRaven/Cataclysm-DDA#35359]
# Describe the bug

The Generic Guns mod hasn't been updated in a couple of months -- and even then it might not have been complete as of that last update -- and guns have definitely been added since then and not incorporated.

# Steps To Reproduce

1. Start a new world with Generic Guns enabled
2. Observe errors (see below) due to outdated blacklists
3. Bring up debug menu (or look at the details in #34236 )
4. Search for `TAC-50` (or `tac50` in json)
5. Observe gun which is not defined in generic guns either as override or blacklist

# Expected behavior

Generic Guns, as an accessibility mod, should remain synchronized with the state of firearms, magazines, and ammunition in vanilla. Given that maintaining Generic Guns requires firearms knowledge, and anyone adding firearms must by definition possess that knowledge, I would argue it should be incumbent on contributors adding or updating guns to ensure Generic Guns is also updated.

# Screenshots

outdated blacklist definition 
![image](https://user-images.githubusercontent.com/1569754/68308815-5456ef80-0062-11ea-814d-f8bbf6f06165.png)

Sum total of `tac50` references in master as of this writing:
![image](https://user-images.githubusercontent.com/1569754/68309396-38a01900-0063-11ea-88de-883130fa237d.png)

# Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 19.10; Release: 19.10; Codename: eoan; 
- Game Version: 0.D-9180-gfcbe9b840b [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Generic Guns [generic_guns]
]

# Additional information

Anyone working on this task might want to be aware of consistency check issues (see: #20320, found by way of  #31321 )

**Comment by tenmillimaster** [comment]
I don't mind doing taking a look at this.

**Comment by RickyVanz** [comment]
im the contributor of Tac-50 and other's .50 sniper rifle. and i agree with this.
